compiler/rustc_middle/src/ty/util.rs

```diff
@@ -1539,16 +1539,28 @@ pub fn is_doc_notable_trait(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
 }
 
 /// Determines whether an item is an intrinsic by Abi.
-pub fn is_intrinsic(tcx: TyCtxt<'_>, def_id: LocalDefId) -> bool {
-    matches!(tcx.fn_sig(def_id).skip_binder().abi(), Abi::RustIntrinsic | Abi::PlatformIntrinsic)
+pub fn intrinsic(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<Symbol> {
+    if matches!(tcx.fn_sig(def_id).skip_binder().abi(), Abi::RustIntrinsic | Abi::PlatformIntrinsic)
+    {
+        Some(tcx.item_name(def_id.into()))
+    } else if let Some(name) = tcx.get_attr(def_id, sym::rustc_intrinsic) {
+        let name =
+            name.get_normal_item().meta(rustc_span::DUMMY_SP).unwrap().meta_item_list().unwrap()[0]
+                .ident()
+                .unwrap()
+                .name;
```

Comment:
Can we maybe get the name of the item instead?

Writing
```rust
#[rustc_intrinsic(blah_blah_blah)]
fn blah_blah_blah() {}
```
Is not _bad_, but I don't see why we can't just write 

```rust
#[rustc_intrinsic]
fn blah_blah_blah() {}
```

Reply:
What existing functions would we want to apply this to?

Yeah, defaulting to to function name sounds good :)

Reply:
I'd assume all of those that just forward to an intrinsic internally. At least that is what was discussed on the issue. It may be cleaner to just have a single very generic function tho